### ISSUE TITLE
LocationRequest deprecated

### ISSUE THREAD

**Issue by sisalik1** [Baseflow/flutter-geolocator#1206]
Hi!

```
flutter run --debug                                   
Launching lib\main.dart on SM A528B in debug mode...
...\Pub\Cache\hosted\pub.dartlang.org\geolocator_android-4.1.7\android\src\main\java\com\baseflow\geolocator\location\FusedLocationClient.java:86: warning: [deprecation] create(
) in LocationRequest has been deprecated
    LocationRequest locationRequest = LocationRequest.create();
                                                     ^
...\Pub\Cache\hosted\pub.dartlang.org\geolocator_android-4.1.7\android\src\main\java\com\baseflow\geolocator\location\FusedLocationClient.java:89: warning: [deprecation] setPrio
rity(int) in LocationRequest has been deprecated
      locationRequest.setPriority(toPriority(options.getAccuracy()));
                     ^
...\Pub\Cache\hosted\pub.dartlang.org\geolocator_android-4.1.7\android\src\main\java\com\baseflow\geolocator\location\FusedLocationClient.java:90: warning: [deprecation] setInte
rval(long) in LocationRequest has been deprecated
      locationRequest.setInterval(options.getTimeInterval());
                     ^
...\Pub\Cache\hosted\pub.dartlang.org\geolocator_android-4.1.7\android\src\main\java\com\baseflow\geolocator\location\FusedLocationClient.java:91: warning: [deprecation] setFast
estInterval(long) in LocationRequest has been deprecated
      locationRequest.setFastestInterval(options.getTimeInterval() / 2);
                     ^
...\Pub\Cache\hosted\pub.dartlang.org\geolocator_android-4.1.7\android\src\main\java\com\baseflow\geolocator\location\FusedLocationClient.java:92: warning: [deprecation] setSmal
lestDisplacement(float) in LocationRequest has been deprecated
      locationRequest.setSmallestDisplacement(options.getDistanceFilter());
                     ^
5 warnings
...\Pub\Cache\hosted\pub.dartlang.org\google_maps_flutter_android-2.4.3\android\src\main\java\io\flutter\plugins\googlemaps\Convert.java:27: warning: [deprecation] FlutterMain i
n io.flutter.view has been deprecated
import io.flutter.view.FlutterMain;
                      ^
1 warning

```

Found a fix, but not quite sure, how to do that (can't find the file):
https://stackoverflow.com/questions/66489605/is-constructor-locationrequest-deprecated-in-google-maps-v2/70649104#70649104

**Comment by hansajith1998** [comment]
```
..\AppData\Local\Pub\Cache\hosted\pub.dev\geolocator_android-4.2.0\android\src\main\java\com\baseflow\geolocator\location\FusedLocationClient.java:86: warning: [deprecation] create() in LocationRequest has been deprecated
    LocationRequest locationRequest = LocationRequest.create();
                                                     ^
..\AppData\Local\Pub\Cache\hosted\pub.dev\geolocator_android-4.2.0\android\src\main\java\com\baseflow\geolocator\location\FusedLocationClient.java:89: warning: [deprecation] setPriority(int) in LocationRequest has been deprecated
      locationRequest.setPriority(toPriority(options.getAccuracy()));
                     ^
..\AppData\Local\Pub\Cache\hosted\pub.dev\geolocator_android-4.2.0\android\src\main\java\com\baseflow\geolocator\location\FusedLocationClient.java:90: warning: [deprecation] setInterval(long) in LocationRequest has been deprecated
      locationRequest.setInterval(options.getTimeInterval());
                     ^
..\AppData\Local\Pub\Cache\hosted\pub.dev\geolocator_android-4.2.0\android\src\main\java\com\baseflow\geolocator\location\FusedLocationClient.java:91: warning: [deprecation] setFastestInterval(long) in LocationRequest has been deprecated
      locationRequest.setFastestInterval(options.getTimeInterval() / 2);
                     ^
..\AppData\Local\Pub\Cache\hosted\pub.dev\geolocator_android-4.2.0\android\src\main\java\com\baseflow\geolocator\location\FusedLocationClient.java:92: warning: [deprecation] setSmallestDisplacement(float) in LocationRequest has been deprecated
      locationRequest.setSmallestDisplacement(options.getDistanceFilter());
                     ^
..\AppData\Local\Pub\Cache\hosted\pub.dev\geolocator_android-4.2.0\android\src\main\java\com\baseflow\geolocator\location\LocationMapper.java:31: warning: [deprecation] isFromMockProvider() in Location has been deprecated
      position.put("is_mocked", location.isFromMockProvider());
                                        ^
6 warnings
..\AppData\Local\Pub\Cache\hosted\pub.dev\geocoding_android-2.1.2\android\src\main\java\com\baseflow\geocoding\Geocoding.java:36: warning: [deprecation] getFromLocationName(String,int) in Geocoder has been deprecated
        return geocoder.getFromLocationName(address, 5);
                       ^
..\AppData\Local\Pub\Cache\hosted\pub.dev\geocoding_android-2.1.2\android\src\main\java\com\baseflow\geocoding\Geocoding.java:54: warning: [deprecation] getFromLocation(double,double,int) in Geocoder has been deprecated
        return geocoder.getFromLocation(latitude, longitude, 5);

```

**Comment by LavanDroid** [comment]
**Flutter**

Launching lib/main.dart on Android SDK built for x86 in debug mode...
main.dart:1
Note: .../.pub-cache/hosted/pub.dev/**geocoding_android-2.1.2**/android/src/main/java/com/baseflow/geocoding/Geocoding.java uses or overrides a **deprecated API**.

Any update?

**Comment by mvanbeusekom** [comment]
Hi @sisalik1, @hansajith1998 and @LavanDroid,

This issue should be solved with version 4.2.1 of the [geolocator_android](https://pub.dev/packages/geolocator_android/changelog) package.
